enterprise/backend/src/metabase_enterprise/metabot_v3/client.clj

```diff
@@ -204,3 +210,53 @@
                               (:status response) (:reason-phrase response))
                       {:request (assoc options :body body)
                        :response response})))))
+
+(defn- build-multipart-request-options
+  "Build request options for multipart upload."
+  [image]
+  (let [base-headers (dissoc (request-headers) "Content-Type")]
+    {:headers          base-headers
+     :multipart        [{:name "image"
+                         :content (:tempfile image)
+                         :filename (:filename image)}]
+     :follow-redirects true
+     :throw-exceptions false}))
+
+(mu/defn analyze-chart
+  "Ask the AI service to analyze a chart image."
+  [chart-data :- [:map
+                 [:image_base64 :string]
+                 [:chart {:optional true} [:map
+                          [:name {:optional true} [:maybe :string]]
+                          [:description {:optional true} [:maybe :string]]]]
+                 [:timeline_events {:optional true} [:sequential [:map
+                                                                 [:name :string]
+                                                                 [:description {:optional true} [:maybe :string]]
+                                                                 [:timestamp :string]]]]]]
```

Comment:
This is hefty, probably worth defining as a standalone named var. Same with the spec for `dashboard-data`